torchbenchmark/__init__.py

```diff
@@ -14,6 +15,31 @@
 model_dir = 'models'
 install_file = 'install.py'
 
+# Enum class to hold all the available domain names
+class Domain(Enum):
+    COMPUTER_VISION = "computer vision"
+    OTHER_COMPUTER_VISION = "other computer vision"
+    NLP = "natural language processing"
+    SPEECH = "speech"
+    RECOMMENDATION = "recommendation"
+    REINFORCEMENT_LEARNING = "reinforcement learning"
+    OTHER = "other"
+
+# Enum class to hold all the available task names
+class Task(Enum):
```

Comment:
i'm wondering if there is a way to organize these hierarchically so that the relationship of Domain->Task is implied by the structure.

It works OK as is, but it's a little annoying that someone _could_ mix NLP domain with OTHER_RL task, for example.

What if you just flattened it - e.g.

class COMPUTER_VISION(Enum):
   SEGMENTATION = ...
   
class NLP(Enum):
   OTHER_NLP = ...
   
this way, you'd only have to set one value per model and it'd capture both parts.

you could also move these enums into a new file torchbenchmark/tasks.py 
so that in each model file you'd be able to 
from torchbenchmark.tasks import COMPUTER_VISION
and then 
self.task = COMPUTER_VISION.SEGMENTATION

Reply:
Yea, I think this makes sense. I just did such a mistake as well. So this is a better design. I'll fix this